amt.go

```diff
@@ -295,6 +306,50 @@ func (n *Node) forEachAt(ctx context.Context, bs cbor.IpldStore, height int, sta
 
 }
 
+func (n *Node) parForEachAt(ctx context.Context, bs cbor.IpldStore, height int, start, offset uint64, parNum int, cb func(uint64, *cbg.Deferred) error) error {
+	errCh := make(chan struct{})
+	toProcess := make(chan struct{}, parNum)
+	for i := 0; i < parNum; i++ {
+		toProcess <- struct{}{}
+	}
+
+	var wg sync.WaitGroup
+	multiErr := &MultiError{}
+	wrapCh := func(u uint64, deferred *cbg.Deferred) error {
+		select {
```

Comment:
Somewhere in here, I would expect to see a check for context cancellation.

I'm also not sure why there's several `select` blocks.